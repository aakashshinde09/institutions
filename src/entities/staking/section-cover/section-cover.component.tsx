import CoverBg from '@public/assets/images/stake-cover.png'
import { cn } from '@shared/lib/utils'
import { motion } from 'framer-motion'

interface ISectionCoverProperties {
  id: string
  title: string
  subtitle?: string
  onClick: () => void
  className?: string
}

export const SectionCover = ({
  id,
  title,
  subtitle,
  onClick,
  className,
}: ISectionCoverProperties) => {
  return (
    <motion.div
      id={id}
      className={cn(
        'relative h-full cursor-pointer overflow-hidden rounded-4xl p-6 w-full',
        className,
      )}
      whileHover="animate"
      onClick={onClick}
    >
      <motion.div
        className="relative z-10"
        style={{ color: '#000' }}
        variants={{
          animate: {
            color: '#fff',
            transition: {
              delay: 0.25,
            },
          },
        }}
      >
        <h3 className="h3">{title}</h3>
        <p className="mt-2 text-base font-bold">{subtitle}</p>
      </motion.div>
      <motion.img
        src={CoverBg.src}
        alt="cover"
        className="absolute bottom-0 left-16 z-10 aspect-square w-[31.25rem]"
        variants={{
          initial: {
            opacity: 1,
          },
          animate: {
            opacity: 0.2,
          },
        }}
      />
      <motion.svg
        height="20"
        width="20"
        className="absolute -bottom-5 -right-5 z-0 fill-primary"
        variants={{
          initial: {
            scale: 1,
            opacity: 0,
          },
          animate: {
            scale: 90,
            opacity: 1,
            transition: {
              duration: 0.5,
              ease: 'easeInOut',
            },
          },
        }}
      >
        <circle cx="10" cy="10" r="10" />
      </motion.svg>
    </motion.div>
  )
}
