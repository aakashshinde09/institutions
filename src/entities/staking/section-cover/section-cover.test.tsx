import { expect, test, describe } from 'vitest'
import { render } from '@testing-library/react'
import { SectionCover } from './section-cover.component'

describe('<SectionCover/>', () => {
  test('Renders correctly', () => {
    const component = render(
      <SectionCover id="2" onClick={() => {}} title="Impact staking" />,
    )

    expect(component).toMatchSnapshot()
  })
})
