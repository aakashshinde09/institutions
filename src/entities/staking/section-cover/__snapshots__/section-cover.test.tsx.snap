// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<SectionCover/> > Renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="relative h-full cursor-pointer overflow-hidden rounded-4xl p-6 w-full"
        id="2"
      >
        <div
          class="relative z-10"
          style="color: rgb(0, 0, 0);"
        >
          <h3
            class="h3"
          >
            Impact staking
          </h3>
          <p
            class="mt-2 text-base font-bold"
          />
        </div>
        <img
          alt="cover"
          class="absolute bottom-0 left-16 z-10 aspect-square w-[31.25rem]"
        />
        <svg
          class="absolute -bottom-5 -right-5 z-0 fill-primary"
          height="20"
          width="20"
        >
          <circle
            cx="10"
            cy="10"
            r="10"
          />
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="relative h-full cursor-pointer overflow-hidden rounded-4xl p-6 w-full"
      id="2"
    >
      <div
        class="relative z-10"
        style="color: rgb(0, 0, 0);"
      >
        <h3
          class="h3"
        >
          Impact staking
        </h3>
        <p
          class="mt-2 text-base font-bold"
        />
      </div>
      <img
        alt="cover"
        class="absolute bottom-0 left-16 z-10 aspect-square w-[31.25rem]"
      />
      <svg
        class="absolute -bottom-5 -right-5 z-0 fill-primary"
        height="20"
        width="20"
      >
        <circle
          cx="10"
          cy="10"
          r="10"
        />
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
